### PR TITLE
Kotlin to date period

### LINKED ISSUE #3819: [Term Entry] Kotlin Datetime .toDatePeriod()
### Reason/inspiration (optional)

We would like a new term entry on the `.toDatePeriod()` method for Kotlin: [Datetime](https://www.codecademy.com/resources/docs/kotlin/datetime). The entry should go in a new file under `docs/content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by Anudhyan**
### Description

- Added a new term toDatePeriod() under Kotlin programming language

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).
Fixes #3819

### CODE REVIEW

**Comment by Anudhyan** [comment]
@ishg-153 @dakshdeepHERE @shvuuuu please give it a first review

**Comment by dakshdeepHERE** [comment]
Hey @Anudhyan please refrain from tagging all the maintainers. They might be busy with some other PRs. Don't worry we will always process your PRs ASAP!! 🚀

**Comment by Anudhyan** [comment]
> Hey @Anudhyan please refrain from tagging all the maintainers. They might be busy with some other PRs. Don't worry we will always process your PRs ASAP!! 🚀

I apologize for my mistake

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:19]:
```suggestion
The **`toDatePeriod`** function parses the ISO-8601 duration representation as a `DatePeriod`. 
The ISO-8601 duration representation is a string that starts with the letter P followed by a sequence of numbers and units. 
The numbers represent the number of units, and the units can be days, weeks, months, years, hours, minutes, or seconds.
The **`toDatePeriod`** function parses ISO-8601 duration representations and returns a `DatePeriod` object. ISO-8601 duration representation is a string that starts with the letter `P` followed by a sequence of numbers and units. The numbers represent the number of units, and the units can be days, weeks, months, years, hours, minutes, or seconds. For example, the following ISO-8601 duration representations represent one day, one week, one month, and one year, respectively.
```

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:21]:
```suggestion

The function returns a `DatePeriod` object with two properties:
- *start*: the start date of the duration.
- *endInclusive*: the end date of the duration, inclusive.
```

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:31]:
```suggestion
The following code creates a `DatePeriod` object representing one day:
```

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:10]:
```suggestion
 - 'Methods'
 - 'Data Structures'
 - 'Time'
```
Check valid [tags](https://github.com/Codecademy/docs/edit/main/documentation/tags.md) before adding them in meta data.

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:6]:
```suggestion
  - 'Computer Science'
  - 'Code Foundations'
```
Again same issue check valid [subjects](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) before adding them in metadata.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I made some changes please commit them.

**Comment by Anudhyan** [comment]
> @Anudhyan here are the requested changes.

Thanks I have made the changes

**Comment by dakshdeepHERE** [comment]
Looks good for 2nd review @Anudhyan .

**Review by dakshdeepHERE — APPROVED** [review]

**Comment by Anudhyan** [comment]
@SSwiniarski sir please check this

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:16]: You seem to be missing the example referenced in the paragraph.

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:36]: How is this the output? if there's only a variable assignment there's no output to the end user. There should be some code in the example to print out some output to the user. Also, if the DatePeriod references the current date, then yo need to say "output like the following" because it will be different every time the code is run.

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:18]: Move lines 18-20 to after the syntax example.

File: content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/toDatePeriod/toDatePeriod.md:26]: Have a sentence after the code block explaining what `String` is.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Anudhyan I did a second review and had several comments.

**Comment by Anudhyan** [comment]
> @Anudhyan I did a second review and had several comments.

@SSwiniarski sir I have made the changes and now it should be good to go ! Now , I am facing an issue , I started working on this issue on 31st of October & got the pull request accepted by one of the maintainer of codecademy before the hacktoberfest ended ; thereby the pull request got added to my hacktoberfest profile too. It was under 7 day review period and would have been accepted by November 6 but sir yesterday you removed the label and entire PR is now showing not accepted under hacktoberfest website. Is there any way to revert this. I had done everything before the event got ended. Kindly try to add the label back

**Review by SSwiniarski — APPROVED** [review]
@Anudhyan I did one edit. Now it looks good to go.

**Pull request merged**
